book/src/introduction.md

```diff
@@ -1,31 +1,31 @@
+- Generating unsafe bindings:
+In this step, the low-level FFI bindings are created from the supplied `*.gir` files. These are essentially direct calls into the related C library and are unsafe. The resulting crate is typically appended with -sys. The operating mode is `sys`.
 
-- The `*.gir` you need will correspond to the project you want to generate bindings for. You can get them from here or directly on ubuntu website (for example: <http://packages.ubuntu.com/zesty/amd64/libgtk-3-dev>).
+- Generating a safe wrapper:
+Next, another crate for a layer on top of these unsafe (sys) bindings is created, which makes them safe for use in general Rust. The operating mode is `normal`.
 
-- The `*.toml` is what is used to pass various settings and options to [gir] for use when generating the bindings - you will need to write one to suit your needs, for an example you can take a look to gtk-rs/sys/gir-gtk.toml.
+- Checking for missing types/methods:
+The operating mode `not_bound` allows you to see the detected types/methods that will not be generated for whatever reasons.
 
-It operates on 4 different modes:
+- Adding documentation:
+After the safe wrapper is created, gir can even generate the documentation for us. Use the operating mode `doc` to do so.
 
-- `sys`: is what creates the low-level FFI bindings from the supplied `*.gir` file - these are essentially direct calls in to the related C library and are typically unsafe. The resulting crate is typically appended with -sys.
 
-- `normal`: generates another crate for a layer on top of these unsafe (sys) bindings which makes them safe for use in general Rust.
-
-- `not_bound`: allows you to see the detected types/methods that will not be generated for whatever reasons.
-
-- `doc`: used for documentation generation
-
-[gir]: https://github.com/gtk-rs/gir
-
-## Helpers
-
-[gir] includes a wrapper script `./generator.py` that detects `Gir.toml` configurations in the current directory (or the path(s) passed on the command-line) and generates "normal" or "sys" crates for it. Alternatively --embed-docs can be passed to prepare source-code for a documentation build by moving all documentation into it. For a complete overview of available options, pass --help.
+## Regenerating the bindings and wrapper
+In order to generate the bindings and the wrapper for the first time, the above mentioned steps should be followed. When you want to regenerate the crates because e.g. the library was updated, you can simplify the process by running the helper script `./generator.py`. The script detects `Gir.toml` configurations in the current directory and sub-directories (or the paths passed on the command-line) and generates "normal" or "sys" crates for it. Alternatively `--embed-docs` can be passed to prepare source-code for a documentation built by moving all documentation into it. For a complete overview of available options, pass `--help`.
```

Comment:
```suggestion
In order to generate the bindings and the wrapper for the first time, the above-mentioned steps should be followed. When you want to regenerate the crates because e.g. the library was updated, you can simplify the process by running the helper script `./generator.py`. The script detects `Gir.toml` configurations in the current directory and subdirectories (or the paths passed on the command-line) and generates "normal" or "sys" crates for it. Alternatively `--embed-docs` can be passed to prepare source-code for a documentation built by moving all documentation into it. For a complete overview of available options, pass `--help`.
```